pmlc/dialect/pxa/transforms/gpu_thread.cc

```diff
@@ -67,12 +67,22 @@ struct GPUThreadPass : public GPUThreadBase<GPUThreadPass> {
       if (!maybeRanges) {
         continue;
       }
-      CostModel model(op, maxThreads);
+      unsigned goalThreads = maxThreads;
+      auto subgroupAttr = op.getAttrOfType<IntegerAttr>("subgroupSize");
+      if (subgroupAttr) {
+        goalThreads = std::max(
+            1u, static_cast<unsigned>(goalThreads / subgroupAttr.getInt()));
+      }
+      CostModel model(op, goalThreads);
       auto tileSize =
           findBestTileSize(EvenTilingGenerator(), model, *maybeRanges);
       auto inner = performTiling(op, tileSize);
       op.setAttr("hardware", StringAttr::get("gpu_block", &getContext()));
       inner.setAttr("hardware", StringAttr::get("gpu_thread", &getContext()));
+      if (subgroupAttr) {
+        op.removeAttr("subgroupSize");
+        inner.setAttr("subgroupSize", subgroupAttr);
```

Comment:
This seems superfluous as we got subgroupAttr from the op attribute "subgroupSize", did not modify it, and now resetting it.

Reply:
We tile the block (i.e. op changes to two block, op = outer, innner = inner).  We are 'moving' the attribute off of the outer block and placing it on the inner block.  In general, I feel like this use of attributes to squirrel information from one pass to another is a poor design, Frank and I have been talking over other options, but haven't arrived at a solution yet.